### PR TITLE
Attempt #2 to add polling for GHA

### PR DESCRIPTION
**Pull request by athornton**

### CODE REVIEW

File: pipelines/sqre/infra/build_sciplatlab.groovy [hunk 1]
```diff
@@ -58,13 +58,42 @@ notify.wrap {
...
+      def url = new URL("https://api.github.com/repos/lsst-sqre/sciplat-lab/actions/workflows/build.yaml/runs?per_page=1&branch=prod")
+
+      def status = ""
+      def conclusion = ""
+      def loop_idx = 0
+      while (status != "completed") {
+        loop_idx += 1
+        if (loop_idx > 1) {
+          sleep(10 * 1000)  // Good old sleep 10 (but not the first time)
+        }
```
Comment (ktlim) [inline pipelines/sqre/infra/build_sciplatlab.groovy:80]: Maybe abort if `loop_idx` gets too big (GHA is stuck for some reason)?

File: pipelines/sqre/infra/build_sciplatlab.groovy [hunk 2]
```diff
@@ -58,13 +58,42 @@ notify.wrap {
...
+        def conn = url.openConnection().with { conn ->
+          conn.setRequestMethod('GET')
+          conn.setRequestProperty('Content-Type', 'application/json; charset=utf-8')
+          conn.setRequestProperty('Accept', 'application/vnd.github.v3+json')
+          def text = conn.getInputStream().getText()
+          def jsonSlurper = new groovy.json.JsonSlurper()
```
Comment (ktlim) [inline pipelines/sqre/infra/build_sciplatlab.groovy:86]: Wondering if this needs to be constructed each time or if it can be reused.

**Review by ktlim — APPROVED** [review]
OK to merge as is, but a couple more ideas for improvements.

File: pipelines/sqre/infra/build_sciplatlab.groovy (hunk 1 above)
Comment (athornton) [reply pipelines/sqre/infra/build_sciplatlab.groovy:80]: OK, I set it to 2 hours.  I should probably also put a timeout in the action itself.  It takes about 25 minutes, usually.  But it's also very dependent on how the conda/mamba solve goes within the container build.

File: pipelines/sqre/infra/build_sciplatlab.groovy (hunk 2 above)
Comment (athornton) [reply pipelines/sqre/infra/build_sciplatlab.groovy:86]: Looks like I can factor it out of the loop.

**Pull request merged**